Fix Irene bug caused by Numpy upgrade

jacg opened the pull request:
We were using Numpy 1.14.5. 1.15.0 is available, but `test_irene_split_trigger` breaks if we use the new version of Numpy.

jacg committed:
Add Python 3.7 to Travis matrix

jacg committed:
Allow Python 3.7 builds to fail on Travis

We are not relying on Python 3.7 yet. It's there just so that we can
see when all our required packages become available.

jacg committed:
Add 3.7 to bash completion options

jacg committed:
Bump versions not available in Python 3.7

The old versions of hypothesis, pandas and pymysql are not available
in the standard conda channel for Python 3.7, so these need to be
bumped for Python 3.7 to work at all.

jacg committed:
Bump versions which don't break anything

Bumping Numpy from 1.14.5 to 1.15.0 breaks `test_irene_split_trigger`,
so we'll keep that one back, and deal with it in a separate PR.

Cython, notebook, pytest and sphinx are available in their old
versions for Python 3.7, but me might as well take this opportunity to
keep up with the latest and greatest.

jacg committed:
Bump Numpy version, breaking test_irene_split_trigger

We were using Numpy 1.14.5. 1.15.0 is available, but
`test_irene_split_trigger` breaks if we use the new version of
Numpy. This commit demonstrates that fact.

jacg committed:
Tolerate warnings which don't concern us

`test_irene_split_trigger` used to check that exactly one
warning (with a very specific message) arose is a given
situation. When other, orthogonal, warnings appeared, the test failed.

The solution is to ensure that there is at least (rather than exactly)
one warning, and to check that the required message appears exactly
once in the collection of warnings that do appear.

jmbenlloch reviewed (approved):
You're right. This is approved, the bug has been successfully fixed.

jjgomezcadenas committed:
493 Keep an eye on Python 3.7

[author: jacg]

Added Python 3.7 as an allowed failure to the Travis build matrix. Like this we will know when all our dependencies are available on conda in Python 3.7.

I have also taken this opportunity to:

+ Update the pinned versions of our dependencies to the most recent ones

+ Update conda to the most recent version, and change the acivation
  command to the one which has been available and recommended since
  4.4 (#455).

[reviewer: jjgomezcadenas]

python 3.7 passes all tests in my machine. As an extra check I have
run a complex analysis notebook that performs all kind of fits to the
lifetime of the detector and the results are identical in 3.6 and
3.7. Thus, it seems that the upgrade is a success.

I recommend keeping both versions of python for a short while. The
current software in production is very stable and I am reluctant to
introduce any changes which are not needed in the prod version, but
the dev version of the software can already go to the bumped-package,
3.7 python. Over the next couple of weeks we can launch a test
production of kdsts with dev and verify that everything is fine.

However I would recommend all developers to move to the bumped 3.7
version, so that once we move the dev (3.7) to prod, we don't need to
support 3.6 anymore (don't see the point, once we have checked
everything is fine and transition has been completed.
